### DISCUSSION TITLE
Graphics folder asset depending on language setting

### DISCUSSION THREAD

**Discussion by SwedishRabbit** [fvtt-fria-ligan/forbidden-lands-foundry-vtt#304] [Ideas]
Hi,
I have an idea - dont know how hard it is to set up:
Since depending on which language you have for your game - the graphics on maps, stickers, artifact- and combat cards, mount-cards, and adventure site art is at the moment limited (locked to english).

I have succesfully replaced my "assets" folder for Core Game and Ravens Purge with an identical structure down to the namings - containging swedish graphic images. So Foundry see no difference when referring, but we as users see the swedish graphic.

Here is my vision:
Could Foundry VTT be prepped with multiple assets folders?
So when the language is determined in Foundry. The target folder for "assets" with all its subfolders is pointed to a specific folder?

for instance Foundry could be prepped with parralel folders:
"assets en" - english graphic
"assets sv" - swedish graphic
"assets fr" - french graphic
"assets es" - spanish graphic
etc...  

And just a thought - some graphics are language independent so there the pointing folder could remain fixed on the original "assets"

If possible a pilot with the swedish graphic could be made since i have one assets folder prepared.

Any comments or thoughts regarding this? Do you think it is possible?

**Comment by aMediocreDad** [comment]
Hi! I will assume this question is with regards to the official modules, and not the system itself.

In short. I'm afraid this isn't possible.

Without being too technical about it, it boils down to the fact that all the urls that point to the assets are hardcoded in storage. The values of these cannot be interpolated later either (in some places some hacky code could work, but it would not be worth it). To make it work would require rewriting how Foundry handles the assets (e.g. when rendering out the canvas).

The only solution is to do what you did; replace all the assets with your own. If the modules were to be released in other languages, this would be the solution (a completely separate module with its own assets).
